File: src/main/java/supermetroid/memory/MemoryReader.java
package supermetroid.memory;

public abstract class MemoryReader {
  public abstract byte readByte(int offset);
  public byte[] readBytes(int offset, byte[] bytes) {
    for (var i = 0; i < bytes.length; i++) {
      bytes[i] = readByte(offset + i);
    }
    return bytes;
  }
  public abstract int readShort(int offset);
  public void close() {}
}

File: src/main/java/supermetroid/memory/MemoryReader.kt
package supermetroid.memory

abstract class MemoryReader {
    abstract fun readByte(offset: Int): Int
    open fun readBytes(offset: Int, bytes: IntArray): IntArray {
        for (i in bytes.indices) {
            bytes[i] = readByte(offset + i)
        }
        return bytes
    }

    abstract fun readShort(offset: Int): Int
    open fun close() {}
}
